### PR TITLE
Added directory and code to `examples\` to reproduce Figure 2 with S.c and S.p

### PR DESCRIPTION
**Pull request by theJasonFan**
# Homologs are more similar in HANDL space than non-homologs

In this example, we show that HANDL scores are correlated with functional similarity by reproducing Figure 2a and Figure 2b (as well as Figure S1 and S2 in the S.I).

### Usage

To produce Figure 2 from [REF] simply execute `snakemake all --configfile configs/<configfile>`

### Provided configuration files:

We provide example configuration files for the following:
* `configs/sc-sp.yml` is a configuration file to produce plots of HANDL dissimilarity scores between _S.c_ and _S.p_ proteins projected into HANDL space with _S.c_ as the source and _S.p_ as the target, respectively. PPI networks from which HANDL scores and embeddings are computed can be found in the `data` directory in the root folder and are obtained from BioGrid v3.4.157.
* `configs/human-sc.yml`, `configs/human-mouse.yml`, `configs/mouse-sc.yml`, are a configuration files to produce plots of HANDL dissimilarity scores between homolog pairs and other pairs and plots of HANDL vs Resnik scores.

### Data
* Run `snakemake` in `/data` in the root directory to download PPI networks and Resnik scores required to produce plots. (Note that Resnik scores are precomputed and downloaded directly from the UMIACS Object Store.)

### CODE REVIEW

File: example/dissim-plot/plot_kde.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    for i, (xs, label) in enumerate(scores_and_labels):
+        scores = np.ravel(xs) if len(xs) < 1e5 else np.random.choice(np.ravel(xs), int(1e5))
+        kde.fit(np.reshape(scores, (-1, 1)))
+        densities = kde.score_samples(X_plot)
+        plt.plot(X_plot[:,0], np.exp(densities), lw = line_width,
+                 label = label, ls=styles[i % len(styles)])
+    plt.ylabel('Density', size = font_size)
+    plt.xlabel(xlabel, size = font_size)
+    plt.legend(loc='best', fontsize = font_size)
+    plt.savefig(output)
```
Comment (mdml) [inline example/dissim-plot/plot_kde.py]: Might add `plt.tight_layout()`

File: example/dissim-plot/plot_kde.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--HANDL_scores', required=True)
+    parser.add_argument('-o', '--output', required=True)
+    parser.add_argument('--xmax', type=float, required=False, default=0.6)
+    return parser.parse_args()
+    
+
+def plot_and_save(scores_and_labels, xlabel, output,
+                  xmin=0.0, 
+                  xmax=0.6,
```
Comment (mdml) [inline example/dissim-plot/plot_kde.py]: Should the `xmax` provided in the config file? Or is this usually the same across species pairs?

File: example/dissim-plot/plot_kde.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    
+    dissims = 1. / sim_scores
+    dissims /= np.mean(dissims)
+    _, _, _, hom_dissims, other_dissims = \
+        handl.separate_scores(dissims, landmark_idxs, homolog_idxs)
+    plots = [(hom_dissims, 'Homologs'), (other_dissims, 'Other')]
+    plot_and_save(plots, 'Dissimilarity scores', args.output)
+
+    
+    # Load
```
Comment (mdml) [inline example/dissim-plot/plot_kde.py]: Not sure what this is...

**Review by mdml — CHANGES_REQUESTED** [review]
A few small changes, but I did run the example successfully.

File: example/dissim-plot/plot_kde.py (hunk 2 above)
Comment (theJasonFan) [reply example/dissim-plot/plot_kde.py]: xmax can be provided in the config file, 0.6 is just a value I picked as the default

File: example/dissim-plot/plot_kde.py (hunk 1 above)
Comment (theJasonFan) [reply example/dissim-plot/plot_kde.py]: Done.

File: example/dissim-plot/plot_kde.py (hunk 2 above)
Comment (theJasonFan) [reply example/dissim-plot/plot_kde.py]: xmax can be provided via the configuration file. There is no obvious way to figure out what xmax should be a priori, so I just picked a default.

File: example/dissim-plot/plot_kde.py (hunk 3 above)
Comment (theJasonFan) [reply example/dissim-plot/plot_kde.py]: Cleaned this up!

**Review by mdml — APPROVED** [review]

**Pull request merged**